### PR TITLE
fix(workers): Fix NuGet config file generation path

### PR DESCRIPTION
**Pull request by MarcelBochtler**
Change path of NuGet config generation to one that complies to the NuGet specification [1].

[1]: https://learn.microsoft.com/en-us/nuget/reference/nuget-config-file

Authored-by: Kamil Bielecki <kamil.bielecki@pl.bosch.com>

### CODE REVIEW

**Review by nnobelis — COMMENTED** [review]
In the linked documentation I don't see the path to `.nuget/Nuget/`.

Quite the opposite:
```
Add a nuget.config file in the root of your project repository. 
```

Can someone explain ?

**Comment by kamil-bielecki-bosch** [comment]
> In the linked documentation I don't see the path `.nuget/Nuget/`.
> 
> Quite the opposite:
> 
> ```
> Add a nuget.config file in the root of your project repository. 
> ```
> 
> Can someone explain ?

There is another NuGet documentation page:
https://learn.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior
In section "On Mac/Linux, the user-level config file location varies by tooling" there is another path advised.

File: workers/common/src/main/kotlin/common/env/NuGetGenerator.kt [hunk 1]
```diff
@@ -30,7 +30,7 @@ import org.eclipse.apoapsis.ortserver.workers.common.env.definition.NuGetDefinit
 class NuGetGenerator : EnvironmentConfigGenerator<NuGetDefinition> {
     companion object {
         /** The name of the configuration file created by this generator. */
-        private const val TARGET = "NuGet.Config"
+        private const val TARGET = ".nuget/NuGet/NuGet.Config"
```
Comment (sschuberth) [inline workers/common/src/main/kotlin/common/env/NuGetGenerator.kt:33]: Should this be made `internal` so it could be used in the tests instead of hard-coding the same string again?

File: workers/analyzer/src/test/kotlin/AnalyzerEndpointTest.kt [hunk 2]
```diff
@@ -287,6 +287,7 @@ class AnalyzerEndpointTest : KoinTest, StringSpec() {
      * should create the configuration files declared for the environment. The specified [test][block] can then
      * check whether these files have been created correctly.
      */
+    @Suppress("LongMethod")
```
Comment (sschuberth) [inline workers/analyzer/src/test/kotlin/AnalyzerEndpointTest.kt:290]: Why is this necessary now if the method was not modified?

File: workers/common/src/main/kotlin/common/env/NuGetGenerator.kt (hunk 1 above)
Comment (kamil-bielecki-bosch) [reply workers/common/src/main/kotlin/common/env/NuGetGenerator.kt:33]: Proper file placement is checked in two kind of tests (unit and integration ones), so that's why this string is used twice.
Also - in case of sharing this value, in fact there should be separate test to check if string content is valid.

**Review by oheger-bosch — APPROVED** [review]
